lib/OpenQA/Worker/Job.pm

```diff
@@ -385,79 +385,76 @@ sub _stop_step_4_upload {
     log_info("Result: $reason",                                  channels => 'autoinst');
 
     # upload logs and assets
-    if ($reason ne WORKER_COMMAND_QUIT && $reason ne 'abort' && $reason ne WORKER_SR_API_FAILURE) {
-
-        Mojo::IOLoop->subprocess(
-            sub {
-                # upload ulogs
-                my @uploaded_logfiles = glob "$pooldir/ulogs/*";
-                for my $file (@uploaded_logfiles) {
-                    next unless -f $file;
-
-                    my %upload_parameter = (
-                        file => {file => $file, filename => basename($file)},
-                        ulog => 1,
-                    );
-                    if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
-                        $reason = WORKER_SR_API_FAILURE;
-                        last;
-                    }
+    if ($reason eq WORKER_COMMAND_QUIT || $reason eq 'abort' || $reason eq WORKER_SR_API_FAILURE) {
+        Mojo::IOLoop->next_tick(sub { $self->_stop_step_5_1_upload($reason, $callback) });
+        return;
+    }
```

Comment:
Since `_stop_step_4_upload` doesn't have a defined return value, you could also do:
```perl
return Mojo::IOLoop->next_tick(sub { $self->_stop_step_5_1_upload($reason, $callback) })
    if $reason eq WORKER_COMMAND_QUIT || $reason eq 'abort' || $reason eq WORKER_SR_API_FAILURE;
```

Reply:
Side note: Since `next_tick` is meant to be used in cases like this it also always returns `undef`.